evalml/automl/utils.py

```diff
@@ -152,3 +154,24 @@ def get_best_sampler_for_data(X, y, sampler_method, sampler_balanced_ratio):
                 return 'SMOTENC Oversampler'
         except ImportError:
             return 'Undersampler'
+
+
+def get_hyperparameter_ranges(component_graph, custom_hyperparameters):
+    """
+    Returns hyperparameter ranges from all components as a dictionary.
+
+    Arguments:
+        component_graph (list): The component_graph of the pipeline.
+        custom_hyperparameters (dict): The custom hyperparameters to be passed to the pipeline.
+
+    Returns:
+        dict: Dictionary of hyperparameter ranges for each component in the component graph.
+    """
+    linearized_component_graph = ComponentGraph.linearized_component_graph(component_graph)
+    hyperparameter_ranges = dict()
+    for component_name, component_class in linearized_component_graph:
+        component_hyperparameters = copy.copy(component_class.hyperparameter_ranges)
+        if custom_hyperparameters and component_name in custom_hyperparameters:
+            component_hyperparameters.update(custom_hyperparameters.get(component_name, {}))
```

Comment:
Nit-pick: I don't think we need the `custom_hyperparameters.get` since we checked that the component name is in the dictionary.